raspiraw.c

```diff
@@ -384,9 +426,19 @@ static void callback(MMAL_PORT_T *port, MMAL_BUFFER_HEADER_T *buffer)
 				file = fopen(filename, "wb");
 				if(file)
 				{
-					if (cfg->write_header)
-						fwrite(brcm_header, BRCM_RAW_HEADER_LENGTH, 1, file);
-					fwrite(buffer->data, buffer->length, 1, file);
+					if (cfg->write_timestamps)
+					{
+						cfg->ptso->idx = count;
+						cfg->ptso->pts = buffer->pts;
+						cfg->ptso->nxt = malloc(sizeof(struct pts_node));
+						cfg->ptso = cfg->ptso->nxt;
```

Comment:
No checking the malloc succeeded -> seg fault.

Simplest solution is change line 429 to
```if (cfg->write_timestamps && cfg->ptso)```
or even 
```if (cfg->ptso)```
as ```cfg->ptso``` will never get set if ```write_timestamps``` is 0.

(Lots of little mallocs like this isn't nice, but such is life. The alternative is to allocate bigger arrays to take eg 1024 timestamps, and chain those together.)